matching/src/main/scala/org/kframework/backend/llvm/matching/pattern/Pattern.scala

```diff
@@ -200,7 +200,7 @@ case class MapP[T](keys: Seq[Pattern[T]], values: Seq[Pattern[T]], frame: Option
   def isSpecialized(ix: Constructor, isExact: Boolean, fringe: Fringe, clause: Clause, maxPriority: Int): Boolean = {
     (ix, frame) match {
       case (Empty(), _) => keys.isEmpty && values.isEmpty
-      case (HasKey(_, _, _), Some(_)) => true
+      case (HasKey(_, _, _), Some(_)) => clause.action.priority <= maxPriority
```

Comment:
I just noticed that this case isn't checking that a choice is occurring before checking the priority. I'm surprised we haven't found any failures running the test suites due to this, but we should still address it.